### PR TITLE
Update gene cache at server side after clicking Update Gene button; Comment bubble update

### LINKED ISSUE #455: Disable the review mode when admin outside of the intranet

### PR DESCRIPTION
**Pull request by zhx828**
Disable review model and Update Gene button if user is not in internal
network;
Close #455, #362, #336

### CODE REVIEW

File: web/yo/app/views/gene.html [hunk 1]
```diff
@@ -9,7 +9,7 @@
       <span ng-if="document" class="geneLastModify" ng-show="gene.status_timeStamp.has('lastEdit')">Last edit was made on {{gene.status_timeStamp.get('lastEdit').value.text | date : 'MMM d, h:mm a yyyy'}} by {{gene.status_timeStamp.get('lastEdit').by.text }}</span>
       <span ng-if="document && userRole === 8" class="geneLastModify" ng-show="gene.status_timeStamp.has('lastUpdate')">Last update to database was made on {{gene.status_timeStamp.get('lastUpdate').value.text  | date : 'MMM d, h:mm a'}} by {{gene.status_timeStamp.get('lastUpdate').by.text }}</span>
       <span class="btn-group btn-group-sm pull-right" role="group">
-        <button type="button" class="btn btn-default" ng-if="userRole === 8" ng-disabled="!fileEditable || !metaDocStatus.saved" ng-click="review()">{{reviewMode ? 'Review Complete' : 'Review'}}</button>
+        <button type="button" class="btn btn-default" ng-if="userRole === 8 && $root.internal" ng-disabled="!fileEditable || !metaDocStatus.saved" ng-click="review()">{{reviewMode ? 'Review Complete' : 'Review'}}</button>
```
Comment (JiaoJiao123) [inline web/yo/app/views/gene.html]: It should be $rootScope.internal?

File: web/yo/app/views/gene.html [hunk 2]
```diff
@@ -565,7 +565,7 @@ <h4 ng-if="!tumor.trials_review.get('action') && !tumor.trials_review.get('remov
     <!--<span ng-if="!status.hideAllEmpty">Hide</span><span ng-if="status.hideAllEmpty">Show</span> all empty-->
   <!--</button>-->
   <!--<button type="button" class="btn btn-default" ng-if="userRole === 8" ng-click="getData()">Check gene data</button>-->
-  <button ng-style="{color: updateGeneColor()}" type="button" class="btn btn-default" ng-if="userRole === 8" ng-click="updateGene()"><span ng-show="!!!docStatus.savedGene"><i class="fa fa-spinner fa-spin"></i></span><span ng-show="!!docStatus.savedGene">Update gene</span></button>
+  <button ng-style="{color: updateGeneColor()}" type="button" class="btn btn-default" ng-if="userRole === 8 && $root.internal" ng-click="updateGene()"><span ng-show="!!!docStatus.savedGene"><i class="fa fa-spinner fa-spin"></i></span><span ng-show="!!docStatus.savedGene">Update gene</span></button>
```
Comment (JiaoJiao123) [inline web/yo/app/views/gene.html]: It should be $rootScope.internal?

File: web/yo/app/views/gene.html [hunk 3]
```diff
@@ -9,7 +9,7 @@
       <span ng-if="document" class="geneLastModify" ng-show="gene.status_timeStamp.has('lastEdit')">Last edit was made on {{gene.status_timeStamp.get('lastEdit').value.text | date : 'MMM d, h:mm a yyyy'}} by {{gene.status_timeStamp.get('lastEdit').by.text }}</span>
       <span ng-if="document && userRole === 8" class="geneLastModify" ng-show="gene.status_timeStamp.has('lastUpdate')">Last update to database was made on {{gene.status_timeStamp.get('lastUpdate').value.text  | date : 'MMM d, h:mm a'}} by {{gene.status_timeStamp.get('lastUpdate').by.text }}</span>
       <span class="btn-group btn-group-sm pull-right" role="group">
-        <button type="button" class="btn btn-default" ng-if="userRole === 8" ng-disabled="!fileEditable || !metaDocStatus.saved" ng-click="review()">{{reviewMode ? 'Review Complete' : 'Review'}}</button>
+        <button type="button" class="btn btn-default" ng-if="userRole === 8 && $root.internal" ng-disabled="!fileEditable || !metaDocStatus.saved" ng-click="review()">{{reviewMode ? 'Review Complete' : 'Review'}}</button>
         <button type="button" class="btn btn-default" ng-if="developerCheck()" ng-click="exitReview()">Exit Review</button>
```
Comment (JiaoJiao123) [inline web/yo/app/views/gene.html]: should we also have this intranet check for "Exit Review" button?

Comment (zhx828) [inline web/yo/app/views/gene.html]: Don't think so. This is the helper to manually exit the review mode. We should be able to do that outside of intranet

File: web/yo/app/views/gene.html (hunk 2 above)
Comment (zhx828) [inline web/yo/app/views/gene.html]: $root is actually referring to $rootScope. But yeah, we should try to only use rootScope. I will change that. Thanks.

**Pull request merged**